### PR TITLE
Estebanreyl/container registry runtime sdk

### PR DESCRIPTION
**Pull request by estebanreyl**
Should finally address : https://github.com/Azure/azure-rest-api-specs/issues/6457

Readying a new release of the container registry runtime sdk along with brand new tests for all methods

History for Review (avoids showing most auto generated files):
https://github.com/Azure/azure-sdk-for-net/pull/7081/files/f957010c20a205aba4aae265a577799394ba9bae..97ff4d756099e72f3034938b720c4eef21b7f4b8?file-filters%5B%5D=.cs

Included:
- An Example program (Would probably be best to break apart and instead use for documentation purposes?)
- Functional Unit tests for 100% of the paths
- A custom authentication file for our own oauth2

### CODE REVIEW

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Helpers/ACRTestUtil.cs [hunk 1]
```diff
@@ -12,37 +12,64 @@ namespace ContainerRegistry.Tests
     using Microsoft.Rest;
     using Microsoft.Rest.ClientRuntime.Azure.TestFramework;
     using Microsoft.Azure.ContainerRegistry;
+    using System.Threading;
```
[the remark this answers is not in the document]
Comment (estebanreyl) [reply sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Helpers/ACRTestUtil.cs]: As in alphabetically?

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs [hunk 2]
```diff
@@ -9,77 +9,239 @@ namespace ContainerRegistry.Tests
...
 
     public class ManifestTests
     {
+        #region Test Values
         private static readonly AcrManifestAttributes ExpectedAttributesOfProdRepository = new AcrManifestAttributes()
         {
-            ImageName = ACRTestUtil.ProdRepository,
-            Registry = ACRTestUtil.ManagedTestRegistryFullName,
-            ManifestAttributes = new AcrManifestAttributesBase()
+            Registry = "azuresdkunittest.azurecr.io",
```
[the remark this answers is not in the document]
Comment (estebanreyl) [reply sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs]: For the overall ExpectedAttributesOfProdRepository like objects or its individual fields?  I can do both as well.

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Helpers/ACRTestUtil.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nathana1) [reply sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Helpers/ACRTestUtil.cs]: yes, visual studio has a command for it

File: sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (nathana1) [reply sdk/containerregistry/Microsoft.Azure.ContainerRegistry/tests/Tests/ManifestTests.cs]: most of the registry names, repositories and tags seem like they are defined in the test utils file so if you can just use those values as appropriate

**Comment by estebanreyl** [comment]
@erich-wang , @isra-fel Any chance I could get an update on what else, if anything, may be missing to get this SDK going?

**Comment by erich-wang** [comment]
@AlexGhiondea, could you take a look since it is data plane related PR?

**Comment by estebanreyl** [comment]
I know its been some time and there's other work but is there any chance we can get this reviewed in the nearby future? V2 is getting really close and its been 2 weeks since the PR was ready.

**Comment by estebanreyl** [comment]
Will close this in favor of releasing V2 since REST spec just went out,

**Pull request closed without merging**